src/main/java/com/google/devtools/build/lib/analysis/SourceManifestAction.java

```diff
@@ -54,7 +54,7 @@
   private static final String GUID = "07459553-a3d0-4d37-9d78-18ed942470f4";
 
   private static final Comparator<Map.Entry<PathFragment, Artifact>> ENTRY_COMPARATOR =
-      (path1, path2) -> path1.getKey().compareTo(path2.getKey());
+      (path1, path2) -> path1.getKey().getPathString().compareTo(path2.getKey().getPathString());
```

Comment:
Interesting, can you elaborate in the PR description why this fixes the test on Windows?

Reply:
Basically we are comparing objects but the tests are comparing path ordering.  Ill add that to the description.

Reply:
OK, this case we actually need it to be case-sensitive to get the correct order, then this change makes sense.